File: business-unit/src/main/java/es/ulpgc/dacd/business/store/EventStoreReader.java
package es.ulpgc.dacd.business.store;

import es.ulpgc.dacd.business.handler.EventHandler;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class EventStoreReader {

    private final String eventStorePath;

    public EventStoreReader(String eventStorePath) {
        this.eventStorePath = eventStorePath;
    }

    public void load(String topic, EventHandler handler) {
        Path topicPath = Path.of(eventStorePath, topic);
        if (!Files.exists(topicPath)) return;
        try {
            Files.walk(topicPath)
                    .filter(p -> p.toString().endsWith(".events"))
                    .sorted()
                    .forEach(file -> loadFile(file, handler));
        } catch (IOException e) {
            System.err.println("[business-unit] Failed to read event store: " + e.getMessage());
        }
    }

    private void loadFile(Path file, EventHandler handler) {
        try (BufferedReader reader = new BufferedReader(new FileReader(file.toFile()))) {
            String line;
            while ((line = reader.readLine()) != null) {
                if (!line.isBlank()) handler.handle(line);
            }
        } catch (IOException e) {
            System.err.println("[business-unit] Failed to read file " + file + ": " + e.getMessage());
        }
    }
}

File: business-unit/src/main/python/controller/event_store_reader.py
import logging
from pathlib import Path

logger = logging.getLogger(__name__)


class EventStoreReader:
    def __init__(self, event_store_path: str):
        self.event_store_path = Path(event_store_path)

    def load(self, topic: str, handler) -> None:
        topic_path = self.event_store_path / topic
        if not topic_path.exists():
            logger.warning(f"Path not found: {topic_path}")
            return

        for file_path in sorted(topic_path.rglob('*.events')):
            self._load_file(file_path, handler)

    def _load_file(self, file_path: Path, handler) -> None:
        try:
            for line in file_path.read_text().splitlines():
                if line.strip():
                    handler(line.strip())
        except Exception as e:
            logger.error(f"Failed to read file {file_path}: {e}")
